fuzzers/main.py

```diff
@@ -0,0 +1,337 @@
+        print(f"Serving command server on port {CMD_PORT}")
+        httpd.serve_forever()
+
+def start_fuzzing(gcs):
+    update_repo('master')
+    cfg = parse_config()
+    print(f"Parsed configuration: {cfg}")
+
+    # Checkout random branch
+    branch = random.choices(cfg['branch'], [b['weight'] for b in cfg['branch']])[0]
+    update_repo(branch['name'])
+
+    # TODO: Minimize the corpus
+    # TODO: Rsync the corpus from gcs more frequently, not just once per fuzzer restart
+    # TODO: Add corpus size gauge metric
+    # TODO: Add coverage metrics (will need to parse logs?)
+    # TODO: Add metrics about overhead (time to update corpus, to build fuzzer, etc.)
+    # TODO: Have one worktree per branch, this would allow both dependency build caching after a branch change and to run fuzzers on multiple branches from a single machine
+    #       The only drawback would be fuzz corpuses wouldn't be shared across processes on the same host on different branches, which is probably not too bad
+
+    # Start the fuzzers
+    for _ in range(NUM_FUZZERS):
+        start_a_fuzzer(branch, cfg, gcs)
+
+def reset_to_gcs(path, local, remote, log_file):
+    print(f"Resetting path {local / path} to GCS {remote}/{path}/")
+    log_file.write(f"Resetting path {local / path} to GCS {remote}/{path}/\n")
+    log_file.flush()
+    #utils.rmdirs(local / path)
+    utils.mkdirs(local / path)
+    #for blob in bucket.list_blobs(prefix = remote + '/' + path + '/'):
+    #    unprefixed_path = blob.name[len(remote) + 1:]
+    #    print(f"Downloading blob {blob.name} to {local / unprefixed_path}")
+    #    blob.download_to_filename(local / unprefixed_path)
+    subprocess.run(
+        ['gsutil', '-m', 'rsync', '-d', f"gs://{GCS_BUCKET}/{remote}/{path}/", local / path],
+        stdout = log_file,
+        stderr = subprocess.STDOUT,
+        check = False, # TODO: currently two concurrent gsutil can step on each other's feet, it still loads a corpus but it ends up with an error
+    )
+
+def auto_upload_to_gcs(exit_event, path, local, remote, bucket, log_file, ignore_deletions, on_uploaded_item, on_deleted_item):
+    print(f"Setting up inotify watch to auto-upload changes to {local / path} to GCS {remote + '/' + path + '/'}")
+    log_file.write(f"Setting up inotify watch to auto-upload changes to {local / path} to GCS {remote + '/' + path + '/'}")
+    log_file.flush()
+
+    def uploader_thread(path, local, remote, bucket, exit_event, log_file):
+        utils.mkdirs(local / path) # Otherwise inotify fails
+        i = inotify.adapters.Inotify()
+        i.add_watch((local / path).as_posix())
+        while True:
+            for event in i.event_gen(yield_nones = False):
+                (_, event_types, _, filename) = event
+                if 'IN_CLOSE_WRITE' in event_types:
+                    log_file.write(f"Uploading new corpus item {local / path / filename} to GCS {remote + '/' + path + '/' + filename}")
```

Comment:
Just store `local / path / filename` and `f'{remote}/{path}/{filename}'` in variables.